readthedocs/core/utils/__init__.py

```diff
@@ -229,7 +229,18 @@ def trigger_build(project, version=None, commit=None):
         # Build was skipped
         return (None, None)
 
-    return (update_docs_task.apply_async(), build)
+    task = update_docs_task.apply_async()
+
+    # FIXME: I'm using `isinstance` here because I wasn't able to mock this
+    # properly when running tests and it fails when trying to save a
+    # `mock.Mock` object in the database.
+    #
+    # Store the task_id in the build object to be able to cancel it later.
+    if isinstance(task.id, (str, int)):
+        build.task_id = task.id
+        build.save()
```

Comment:
@stsewd do you have an idea how we can mock this? At this point `update_docs_task` _and_ `task` are `mock.Mock` objects. It would be good if we can make `task.id` to return an integer or a string. I've tried using `mock.return_value` and others and I wasn't able to make it work.

Reply:
Which test is running this code? Something like `update_docs_task_mock.apply_async.return_value = MagicMock(id=1)` should do it.

maybe `update_docs_task_mock.apply_async().id = 1` could work as well.

Reply:
> Which test is running this code?

Most of the tests from `readthedocs/projects/tests/test_build_tasks.py`. For example, `test_successful_build`

The solution that you suggested is what I've tried and I wasn't able to mock them properly 😞 

I'd appreciate it if you find out the way to mock it. Otherwise, I think I will commit the code with this `if` for now.

Reply:
All tests from that file pass for me. But some tests from `rtd_tests/tests/test_builds.py` fail, but with a mock like this `        update_docs_task.signature().apply_async().id = 1
` they pass

Reply:
OK! I'm merging it for now because I want to deploy this change today. I'll come back to this in the following days. If you have the diff required at hand, can you open a PR with the changes?